drivers/power/max17851/max17851.c

```diff
@@ -0,0 +1,760 @@
+	if (ret)
+		return ret;
+
+	for (i = 0; i < len; i++)
+		data[i] = desc->buff[i + 1];
+
+	desc->lssm_byte = desc->buff[i + 1];
+
+	ret = max17851_clear_buff(desc, false);
+	if (ret)
+		return ret;
+
+	return 0;
+}
+
+/**
+ * @brief Write message over UART using MAX17851.
+ * @param desc - MAX17851 device descriptor.
+ * @param data - Pointer to data containing buffer.
+ * @param len - Length of the message in bytes.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max17851_write_msg(struct max17851_desc *desc, const uint8_t *data,
+		       uint8_t len)
+{
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->buff,
+		.rx_buff = desc->buff,
+		.bytes_number = len + 2,
+		.cs_change = 1,
+	};
+	int ret, i;
+
+	ret = max17851_config_gen2(desc, MAX17851_TX_QUEUE_SEL, true);
+	if (ret)
+		return ret;
+
+	if (len > 31)
+		return -EINVAL;
+
+	memset(desc->buff, 0, len + 2);
+	desc->buff[0] = MAX17851_LDQ_REG;
+	desc->buff[1] = len;
+	for (i = 0; i < len; i++)
+		desc->buff[i + 2] = data[i];
+
+	ret = no_os_spi_transfer(desc->spi_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	desc->buff[0] = MAX17851_NXT_LDQ_REG;
+	xfer.bytes_number = 1;
+
+	ret = no_os_spi_transfer(desc->spi_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	ret = max17851_config_gen2(desc, MAX17851_TX_QUEUE_SEL, false);
+	if (ret)
+		return ret;
+
+	ret = max17851_clear_buff(desc, true);
+	if (ret)
+		return ret;
+
+	return 0;
+}
+
+/**
+ * @brief Configure general settings for MAX17851.
+ * @param desc - MAX17851 device descriptor.
+ * @param sel - Specific setting to be changed.
+ * @param enable - Disable/Enable specific setting.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max17851_config_gen2(struct max17851_desc *desc,
+			 enum max17851_config_gen2_sel sel, bool enable)
+{
+	return max17851_reg_update(desc, MAX17851_CONFIG_GEN2_REG,
+				   MAX17851_CONFIG_GEN2_MASK(sel),
+				   no_os_field_prep(MAX17851_CONFIG_GEN2_MASK(sel), enable));
+}
+
+/**
+ * @brief Set timing for Alert Packets/Keep-Alive Packets.
+ * @param desc - MAX17851 device descriptor.
+ * @param time - Specific pre-defined time value to be set.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max17851_sel_alrtpckt_time(struct max17851_desc *desc,
+			       enum max17851_alrtpckt_time time)
+{
+	return max17851_reg_update(desc, MAX17851_CONFIG_GEN3_REG,
+				   MAX17851_ALRTPCKT_TIME_MASK,
+				   no_os_field_prep(MAX17851_ALRTPCKT_TIME_MASK,
+						   time));
+}
+
+/**
+ * @brief Select queue for transmitting/loading TX buffer for UART.
+ * @param desc - MAX17851 device descriptor.
+ * @param sel - Specific pre-defined value for either TX or LD queue selection.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max17851_select_queue(struct max17851_desc *desc,
+			  enum max17851_queue_sel sel)
+{
+	uint8_t mask;
+
+	mask = (sel / 4) ? MAX17851_TXQ_MASK : MAX17851_LOADQ_MASK;
```

Comment:
`sel >= MAX17851_TXQ_0_SEL` better describes what you're trying to do here. And also, you get rid of the division.